src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/PointerNativeTypeAssignmentRewriter.cs

```diff
@@ -24,6 +24,29 @@ public PointerNativeTypeAssignmentRewriter(string nativeIdentifier, PointerTypeS
             _nativeType = nativeType;
         }
 
+        public override SyntaxNode? VisitVariableDeclarator(VariableDeclaratorSyntax node)
+        {
+            if (node.Initializer is null)
+            {
+                return base.VisitVariableDeclarator(node);
+            }
+
+            if (node.Identifier.ToString() == _nativeIdentifier)
+            {
+                return node.WithInitializer(
+                    EqualsValueClause(
+                        CastExpression(MarshallerHelpers.SystemIntPtrType, node.Initializer.Value)));
+            }
+            if (node.Initializer.Value.ToString() == _nativeIdentifier)
```

Comment:
What is the difference between `Identifier` and `Initializer`?

Reply:
`Identifier` is the name of the variable that's being declared, `Initializer` is the `=  <expression>` where `Initializer.Value` is the expression that's being assigned to the new variable.

Reply:
Basically it's the right vs left members for `VariableDeclarator` in comparison to `AssignmentExpression`